### PR TITLE
buildtool: fail build when user time-traveled to the past

### PR DESCRIPTION
**Pull request by sluongng**
Detect when system clock on client's environment was updated to a time
in the past and fail that build instead of reporting negative build
duration.

---

We have a tenant who have user updating local clock mid-build, resulting in
negative build duration in BES.  Let's not support such time-travel use cases
and fail the build with relevant error message.

I looked at setting up some a test case for this but it resulted in a rather
relatively complex modification of `BlazeRuntimeWrapper` to mock
`BlazeClock.instance()` to modify `runtime.getClock()` call. Given how rare
this case is to reproduce, I don't think it's worth the effort to pursue such
test.

### CODE REVIEW

**Comment by sluongng** [comment]
See https://github.com/bazelbuild/bazel/pull/18301 for similar class of issue.

**Comment by c-mita** [comment]
I do not feel like I'm the best person to review this.

But, I don't think we should be trying to detect changes to the system clock and arbitrarily deciding some changes are worthy of a build failure.

**Comment by sluongng** [comment]
I am open to suggestions here, but currently, if the user changes their local clock mid-build (due to traveling between time zones, daylight saving update, or some other reasons), then the BuildCompleteEvent event will be sent with an invalid time. 

This makes it pretty hard to parse build duration on the server side.

**Comment by sluongng** [comment]
This happened a year ago with one of our users commenting:

> I've confirmed with the developer that he actually changed the date on the computer mid build. 

It's unclear to us on _why_ they did that. 😞

**Comment by meisterT** [comment]
I would argue that reporting negative build duration is not too bad for this should-never-happen cases. So let's rather avoid adding code that deals with it.
